File: tests/fixtures/java/scope.java
package fixtures.scope;

import fixtures.runner.Runner;
import static fixtures.helpers.Tools.run;

class Scope {
    void execute(Runnable run) {
        run.run();

        Runnable callback = this::helper;
        callback.run();

        helper();
        Runner.start();
        run();
        new Worker(build());
    }

    void helper() {}

    Scope copy() {
        return new Scope();
    }
}

File: tests/fixtures/rust/scope.rs
use crate::runner::run;

pub fn execute(run: impl Fn()) {
    run();

    let callback = helper;
    callback();

    {
        use crate::nested::run;
        run();
    }

    {
        late_run();
        use crate::late::late_run;
    }

    helper();
}

fn helper() {}
